tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { test, expect } from "@playwright/test";
+
+test.beforeEach(async ({ page }) => {
+    await page.goto("/");
+    await page.getByText(" Owners").click();
+    await page.locator(".dropdown-menu").getByText(" Search").click();
+    await expect(page.locator("h2")).toHaveText("Owners");
+});
+
+test('Select the desired date in the calendar', async ({ page }) => {
+    await page.getByRole('link', { name: 'Harold Davis' }).click()
+    await page.getByRole('button', { name: 'Add New Pet' }).click()
+
+    const icon = page.locator('span.glyphicon.form-control-feedback').first()
+    await expect(icon).toHaveClass(/glyphicon-remove/);
+    await page.locator('#name').fill('Tom')
+    await expect(icon).toHaveClass(/glyphicon-ok/);
+
+    await page.locator('.mat-mdc-button-touch-target').click()
+
+    const searchYear = '2014';
+    const searchMonth = '05';
+    const searchDay = '02';
+    const dateToAssert = `${searchYear}/${searchMonth}/${searchDay}`
+
+    let calendarMonthAndYear = await page.locator('[aria-label="Choose month and year"]').textContent()
+
+    while (!calendarMonthAndYear?.includes(`${searchMonth} ${searchYear}`)) {
+        await page.locator('[aria-label="Previous month"]').click()
+        calendarMonthAndYear = await page.locator('[aria-label="Choose month and year"]').textContent()
+    }
```

Comment:
Good point)

what about the next approach? 

``` suggestion
    const searchYear = '2014';
    const searchMonth = '05';
    const searchDay = '02';
    const dateToAssert = `${searchYear}/${searchMonth}/${searchDay}`

    await page.getByLabel('Choose month and year').click()
    let calendarHeader = await page.getByLabel('Choose date').textContent()
    let yearFrom = calendarHeader?.split(' – ').map(Number)[0]

    while (yearFrom && (yearFrom > Number(searchYear))) {
        await page.getByLabel('Previous 24 years').click()
        calendarHeader = await page.getByLabel('Choose date').textContent()
        yearFrom = calendarHeader?.split(' – ').map(Number)[0]
    }

    await page.getByText(searchYear).click()
    await page.getByText('May').click()
    await page.locator('[class="mat-calendar-body-cell"]').getByText(searchDay[0] == '0' ? searchDay.slice(1) : searchDay, { exact: true }).click()
    await expect(page.locator('[name="birthDate"]')).toHaveValue(dateToAssert)
```

Reply:
Did you write it or ChatGPT? Be honest ))

Reply:
Well, to be honest - I did. Exept this small part  `.map(Number)`.

About other I think it's obvious - because the logic is confusing, and seems that everything can be simplified somehow

Reply:
Simplification - is what you always have to start with.
Write the simple code first, then refactor to make it pretty. Starting from the other end using ChatGPT you start digging the rabbit hole that you can't get out of when your code fails and you ll need to fix it.

So I would recommend re-writing this by yourself and making it simple and understandable. Then look at your code again, and see how it can be optimized and simplified even more.

Reply:
but it was written by myself) 
ok, will try to rewrite

Reply:
Oh... I am sorry.. looks like I didn't get your answer correctly.

Reply:
no problem) 
in any case, it seems to need to be simplified

Reply:
I did my best 🤞

Reply:
@AlexxDk this one.

Reply:
Now I see, strange.
I thought that once hardcoded date could be changed and in that case no need to update other parts of the code. It is already adapted.
If it is good practice also hardcode steps to achieve 2014 years - no problem. Will do.
Thanks